halibot/halobject.py

```diff
@@ -73,13 +73,30 @@ def sync_send_to(self, msg, dests):
 
 	async def _receive(self, msg):
 		try:
-			self.receive(msg)
+			fname = 'receive_' + msg.type
+			if hasattr(self, fname) and callable(getattr(self, fname)):
+				# Type specific receive function
+				getattr(self, fname)(msg)
+			else:
+				# Generic receive function
+				self.receive(msg)
 		except Exception as e:
 			traceback.print_exc()
 
 	def receive(self, msg):
 		pass
 
+	def receive_help(self, msg):
+		prefix = 'help_'
+		if msg.body == []:
+			# Retrieve available topics
```

Comment:
Is there an `#ifdef` comparable thing we could do in Python? Thinking of the case where maybe we want to dynamically hide unused or disabled features, would there be a way to hide certain functions from being included in this?

Reply:
I don't think there is such a thing... if we want to disable topics, we will have to do it in the case where we are retrieving the list of topics and in the case where where are retrieving the help text for a specific topic. It sounds like a feature to be added later...